Services/Search/classes/Lucene/class.ilLuceneSearchObjectListGUIFactory.php

```diff
@@ -58,7 +53,7 @@ public static function factory(string $a_type) : ilObjectListGUI
         $full_class = "ilObj" . $class . "ListGUI";
 
         include_once($location . "/class." . $full_class . ".php");
-        $item_list_gui = new $full_class();
+        $item_list_gui = new $full_class(ilObjectListGUI::CONTEXT_SEARCH);
```

Comment:
I just did a quick check, if this parameter is correctly passed by the overwritten constructors. This is not always the case, I found: BookingPool, Forum, ItemGroup, LearningSequence, OrgUnit, RootFolder, SAHSLearningModule, Session, StudyProgramme, Survey, RemoteObjectBase.

Actually I am not sure, what CONTEXT_SEARCH controls exactly after quickly looking at the code.

But all these constructors might need to be fixed, or am I wrong?

Reply:
Setting the context to CONTEXT_SEARCH is needed here to make sure that uploading files to container objects that show up in the search results is disabled (the file dropzones are causing an error, and this functionality shouldn't be enabled in the search results anyways).

As far as I can see you are right, all overwritten constructors that don't already pass the context should be fixed to do so.